### DISCUSSION TITLE
react-daisyui

### DISCUSSION THREAD

**Discussion by christianblandford** [saadeghi/daisyui#540] [General]
Hi @saadeghi 

I have begun the process of creating a react wrapper for daisyui. It's very much a work in progress right now (with some bugs), but I have gotten a good amount of components added in just a few days. 

I found [this thread](https://github.com/saadeghi/daisyui/issues/163) and saw that you were okay with the idea of someone doing something like this. I wanted to get your approval before I go about finishing this and releasing/advertising it publicly. I have the code licensed as MIT, so it should be the same as this package. I also tried to mirror your documentation/examples as much as possible, and stay somewhat close to how writing code with daisyui feels without react.

If you are interested, take a look here: [react-daisyui](https://github.com/christianblandford/react-daisyui) or go straight to the [storybook](https://christianblandford.github.io/react-daisyui/) to see the components and how they work. Interested to hear your thoughts.

**Comment by saadeghi** [comment]
Great job 👏  
Yes, as long it's MIT licensed and it mentions the main daisyUI repo, I will support it.  
I've found a few other projects on GitHub trying to make React (or Vue) wrappers for daisyUI but I think they're left incomplete at some point.  
I would be glad to link to it and promote it if it's ready for production.

**Reply by christianblandford** [reply]
Thank you! I have linked directly on the repo/npm page, and will make sure people know this is a derivative of all of the work that you have done.

It would be amazing to be linked :) I'm still trying to get through creating all of the components, but I will likely be pushing a daily update with new components until I am caught up. At the very least, I would love to have some attention on the project, and add a few collaborators, since (as I'm sure you know) maintaining a component library can be a lot of work.

**Comment by benjitrosch** [comment]
Hey @christianblandford @saadeghi, I've been working on a React wrapper as well since December, seems great minds think alike 😆. Only found out yours existed earlier today when I went to go register it on npm and found the name taken!

Here's a link to [mine ](https://github.com/benjitrosch/react-daisy) with the [Storybook](https://benjitrosch.github.io/react-daisy/). Maybe we could combine our efforts somehow? Would make maintaining the componet library a lot easier.

**Reply by christianblandford** [reply]
Hey @benjitrosch , sorry I stole the name out from under you haha. I'm happy to add you as a collaborator to my repo, I'll create a new discussion over there.
